### PR TITLE
make changes in student class

### PR DESCRIPTION
**Pull request by youssefwael184**
1/ update constractors
2/ make removeaccount function 
3/ create login function 
4/ create getprofile function 
5/ bamoooot

### CODE REVIEW

File: TestGeneratorSystem/src/testgeneratorsystem/Student.java [hunk 1]
```diff
@@ -3,28 +3,41 @@
...
     private List<String> takenTests;
-
-    
-    public Student(String grade, int passedTestsCount, double totalTimeOfAllTests, String institute, List<String> favoriteQuestions, List<String> takenTests) {
-        this.grade = grade;
-        this.passedTestsCount = passedTestsCount;
-        this.totalTimeOfAllTests = totalTimeOfAllTests;
+    private static ArrayList<Student> listOfStudents = new ArrayList<>();
+
+    public Student(
```
Comment (mohab-kamle) [inline TestGeneratorSystem/src/testgeneratorsystem/Student.java:19]: common list should exist as parameter in the constructor to represent all the data members of the user class

Comment (youssefwael184) [reply TestGeneratorSystem/src/testgeneratorsystem/Student.java:19]: ماشي هعدله و ابعتلك

File: TestGeneratorSystem/src/testgeneratorsystem/Student.java [hunk 2]
```diff
@@ -100,6 +113,69 @@ public double getAverageScore() {
...
+    public void removeAccount(){
+        try {
+           listOfStudents.remove(this); 
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+        System.out.println("the account is removed");
+    }
+    @Override
+    public Student login(){
```
Comment (mohab-kamle) [inline TestGeneratorSystem/src/testgeneratorsystem/Student.java:126]: remove any system.out.println indicating login success or fail , should be done on the main class

File: TestGeneratorSystem/src/testgeneratorsystem/Student.java [hunk 3]
```diff
@@ -100,6 +113,69 @@ public double getAverageScore() {
...
+        System.out.println("Login successful!");
+        return authenticatedStudent;
+    }
+    else
+    { 
+        System.out.println("Login failed. Please check your username and password.");
+        return null;
+    }
+    }
+    public static Student authenticate(String username, String password) {
```
Comment (mohab-kamle) [inline TestGeneratorSystem/src/testgeneratorsystem/Student.java:143]: needs to be refactored later on to be the same as login function in admin (don't make any current changes)

File: TestGeneratorSystem/src/testgeneratorsystem/Student.java [hunk 4]
```diff
@@ -100,6 +113,69 @@ public double getAverageScore() {
...
+            System.out.println("\nGrade : " + getGrade());
+            System.out.println("\ninstitute : " + getInstitute());
+            System.out.println("\npassedTestsCount : " + getPassedTestsCount());
+            System.out.println("\ntotalTimeOfAllTests : " + getTotalTimeOfAllTests());
+            System.out.println("\nFavorite Questions: ");
+            for (String question : getFavoriteQuestions()) {
+            System.out.println(" - " + question);}
+            System.out.println("\ntakenTests: ");
+            for (String test : getTakenTests()) {
+            System.out.println(" - " + test);}
```
Comment (mohab-kamle) [inline TestGeneratorSystem/src/testgeneratorsystem/Student.java:175]: would be problematic currently as it won't be the best view in the terminal , but won't be a problem in the GUI (the same comment on the question)

**Review by mohab-kamle — APPROVED** [review]

**Pull request merged**